Mathlib/GroupTheory/Perm/ConjAct.lean

```diff
@@ -0,0 +1,63 @@
+/-
+Copyright (c) 2024 Antoine Chambert-Loir. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Chambert-Loir
+-/
+import Mathlib.GroupTheory.GroupAction.ConjAct
+import Mathlib.GroupTheory.Perm.Support
+import Mathlib.GroupTheory.Perm.Cycle.Basic
+import Mathlib.Data.Finset.Pointwise
+
+/-! # Some lemmas pertaining to the action of `ConjAct (Perm α)` on `Perm α`
+
+We prove some lemmas related to the action of `ConjAct (Perm α)` on `Perm α`:
+
+Let `α` be a decidable fintype.
+
+* `conj_support_eq` relates the support of `k • g` with that of `g`
+
+* `cycleFactorsFinset_conj_eq`, `mem_cycleFactorsFinset_conj'`
+  and `cycleFactorsFinset_conj` relate the set of cycles of `g`,  `g.cycleFactorsFinset`,
+  with that for `k • g`
+
+-/
+
+namespace Equiv.Perm
+
+open scoped Pointwise
+
+variable {α : Type*} [DecidableEq α] [Fintype α]
+
+/-- `a : α` belongs to the support of `k • g` iff
+  `k⁻¹ * a` belongs to the support of `g` -/
+theorem conj_support_eq (k : ConjAct (Perm α)) (g : Perm α) (a : α) :
+    a ∈ (k • g).support ↔ ConjAct.ofConjAct k⁻¹ a ∈ g.support := by
+  simp only [mem_support, ConjAct.smul_def, not_iff_not, coe_mul,
+    Function.comp_apply, ConjAct.ofConjAct_inv]
+  apply Equiv.apply_eq_iff_eq_symm_apply
+
+theorem cycleFactorsFinset_conj (g k : Perm α) :
+    (ConjAct.toConjAct k • g).cycleFactorsFinset =
+      Finset.map (MulAut.conj k).toEquiv.toEmbedding g.cycleFactorsFinset := by
+  ext c
+  rw [ConjAct.smul_def, ConjAct.ofConjAct_toConjAct]
+  rw [Finset.mem_map_equiv, ← mem_cycleFactorsFinset_conj_iff k⁻¹]
+  simp only [MulEquiv.toEquiv_eq_coe, MulEquiv.coe_toEquiv_symm, MulAut.conj_symm_apply]
+  group
+
+/-- A permutation `c` is a cycle of `g` iff `k • c` is a cycle of `k • g` -/
+theorem mem_cycleFactorsFinset_conj'
+    (k : ConjAct (Perm α)) (g c : Perm α) :
+    k • c ∈ (k • g).cycleFactorsFinset ↔ c ∈ g.cycleFactorsFinset := by
```

Comment:
I have no experience about that. It looks like the right hand side is clearly simpler than the left hand side, which is of a very specific form, so that having this lemma tagged as simp should do no harm.

Reply:
Let's try…